types/react-native/Codegen.d.ts

```diff
@@ -29,3 +27,32 @@ declare module 'react-native/Libraries/Utilities/codegenNativeComponent' {
 
     export default codegenNativeComponent;
 }
+
+declare module 'react-native/Libraries/Types/CodegenTypes' {
+    import type { NativeSyntheticEvent } from 'react-native';
+
+    // Event types
+    // We're not using the PaperName, it is only used to codegen view config settings
+
+    export type BubblingEventHandler<T, PaperName extends string | never = never> = (
+        event: NativeSyntheticEvent<T>,
+    ) => void | Promise<void>;
+    export type DirectEventHandler<T, PaperName extends string | never = never> = (
+        event: NativeSyntheticEvent<T>,
+    ) => void | Promise<void>;
+
+    // Prop types
+    export type Double = number;
+    export type Float = number;
+    export type Int32 = number;
+    export type UnsafeObject = object;
+
+    type DefaultTypes = number | boolean | string | ReadonlyArray<string>;
+    // Default handling, ignore the unused value
+    // we're only using it for type checking
+    //
+    // TODO: (rickhanlonii) T44881457 If a default is provided, it should always be optional
+    //  but that is currently not supported in the codegen since we require a default
+
+    export type WithDefault<Type extends DefaultTypes, Value extends Type | string | undefined> = Type | undefined;
```

Comment:
Flow's optional type `?T` means `T | undefined | null`, it would be better to have `null` also listed here.

Reply:
Good catch, fixed thanks